app/src/main/java/org/mozilla/focus/settings/SettingsFragment.java

```diff
@@ -139,15 +147,10 @@ public void onResume() {
 
         // Update title and icons when returning to fragments.
         final ActionBarUpdater updater = (ActionBarUpdater) getActivity();
-        final Bundle args = getArguments();
-        final int titleResId = args != null ?
-                args.getInt(TITLE_RESID_INTENT_EXTRA, R.string.menu_settings) :
-                R.string.menu_settings;
-        updater.updateTitle(titleResId);
+        updater.updateTitle(settingsScreen.titleResId);
         updater.updateIcon(R.drawable.ic_back);
 
-        // TODO: #1671 would add Enum types so figuring out the current Settings screen is cleaner
-        if (args != null && args.getInt(PREFERENCES_RESID_INTENT_EXTRA, 0) == R.xml.search_engine_settings_featureflag_manual) {
+        if (settingsScreen == SettingsScreen.SEARCH_ENGINES) {
```

Comment:
Do we also need to check the feature flag here? Before we were only checking one of the values behind SEARCH_ENGINES.